sqlmesh/core/context.py

```diff
@@ -606,12 +620,23 @@ def load(self, update_schemas: bool = True) -> GenericContext[C]:
                     self._models.update({fqn: model.copy(update={"mapping_schema": {}})})
                     continue
 
-            update_model_schemas(self.dag, models=self._models, context_path=self.path)
+            update_model_schemas(
+                self.dag,
+                models=self._models,
+                context_path=self.path,
+                linter=self._linter,
+            )
+
+            if self.config.linter.enabled:
+                self._linter = Linter().fill_from_config(self._linter_rules, self.config.linter)
```

Comment:
We shouldn't need to instantiate a `Linter` to do this.

- Rename `fill_from_config` to `from_rules`
- Make it a class method in `Linter` and handle instantiation in there (`instance = cls(); ...`)

Imo the resulting API should look like `Linter.from_rules(...)`. This pattern is consistently used in the codebase, e.g. `Snapshot.from_node`, `IntervalUnit.from_cron`, etc.